### DISCUSSION TITLE
cordova session management

### DISCUSSION THREAD

**Discussion by sairaja2002** [apache/cordova#480] [Q&A]
Hello, I am using cordova for android. Cordova is unable to save the cookies and it is throwing session expiry. How to save those cookies. I have tried to save cookies from client side but it is not working.

**Comment by smodatos** [comment]
Make Cordova 'htm'-pages with (full-)head and body.
Inside the body (with it's headline, logo etc.), put an empty div with id, which you fill up with fetched content (without html-head, without html-body) from your server.
Your server, yes, knows about cookies, stores them, changes them etc. - read them all, put them in an (encrypted) string, and store this string in the localStorage for further use, which always would be creating serverside code, so before fetch new content from there (which might be affected by cookies), read the localStorage string and send it together with the rest of parameters (e.g. what content to ask for ...) to the server.
There you decrypt the received string - and you get your 'cookies' if they where cookies.
Note: on serverside code (as it's transferred an executed on the device), to save in the localStorage, you may use the onload-function of the "smallest base64 gif" (see there)
